src/app/view/single-picture-view/single-picture-view/single-picture-view.component.ts

```diff
@@ -1,33 +1,80 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, HostBinding, NgZone, OnInit } from '@angular/core';
 import { Location } from '@angular/common';
 import { ImageData } from 'src/app/model/ImageData';
+import { ImageDataContract } from '../../../model/ImageDataContract';
+import { ImageDataFacade } from '../../../model/ImageDataFacade';
+import { DirectoryService } from '../../../service/directory.service';
+import { SnackBarService } from '../../../service/snack-bar.service';
+import { SnackBarType } from '../../../component/single-picutre/snack-bar/snack-bar.component';
 
 @Component({
     selector: 'app-single-picture-view',
     templateUrl: './single-picture-view.component.html',
     styleUrls: ['./single-picture-view.component.scss'],
 })
 export class SinglePictureViewComponent implements OnInit {
+    imageDataFacade?: ImageDataFacade;
     shouldDisplayImgView = false;
-    imgData: ImageData | undefined;
 
-    constructor(private location: Location) {}
+    @HostBinding('class')
+    class = 'view';
+
+    constructor(
+        private location: Location,
+        private directoryService: DirectoryService,
+        private snackBarService: SnackBarService,
+        private ngZone: NgZone,
+    ) {}
+
+    isImageDataContract(object: unknown): object is ImageDataContract {
+        if (object) {
+            const objectAsImageDataContract: ImageDataContract = object as ImageDataContract;
+            return objectAsImageDataContract.name !== undefined && objectAsImageDataContract.path !== undefined;
+        }
+        return false;
+    }
 
     ngOnInit(): void {
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```

Comment:
I was considering such a solution. For sure it would be better in case of passing simple parameter not whole object. According to my knowledge I would have to split it to couple of parameters and additionally stringify it to pass EXIF data as query parameters. That's why I would opt for leaving passing parameters using state.